format/spec.md

```diff
@@ -1754,6 +1754,14 @@ Snapshot summary can include metrics fields to track numeric stats of the snapsh
 | **`engine-name`**        | "spark"    | Name of the engine that created the snapshot                    |
 | **`engine-version`**     | "3.5.4"    | Version of the engine that created the snapshot                 |
 
+### Assignment of Snapshot IDs and `current-snapshot-id`
+
+Writers should produce positive values for snapshot ids in a manner that minimizes the probability of id collisions and should verify the id does not conflict with existing snapshots. Producing snapshot ids based on timestamps alone is not recommended as it increases the potential for collisions.
```

Comment:
is generating a positive value a requirement here?  It seems that it might be more of than an implementation note that "-1" should be reserved because it has special meaning prior to V3?

Reply:
There's nothing in the spec that that explicitly prohibits negative values, but the approach by the reference implementation intentionally avoids negative snapshot ids. There is commentary on this in #57 and #2506, so I think there is an expectation even if not explicitly called out.

Reply:
Right, so I think -1 can't just be an implementation detail with a specific version modifier saying it is ok to produce. if we don't think negative numbers break anything then I think it seems fine to leave as an implementation detail.

Reply:
I don't remember it ever being `System.currentTimeMillis`. Was it really?

Reply:
Yes, the original was just current millis, but it had issues (particularly with tests having collisions).